### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false negatives for `filter`

### ISSUE THREAD

**Issue by henrikhorluck** [stylelint/stylelint#9148]
### What is the problem you're trying to solve?

light-dark is a practical function that can give different colors depending on lightmode or darkmode 
https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Values/color_value/light-dark

However, it only works for _colors_. It can be very tempting to use light-dark for e.g. drop-shadow filters like this: `filter: light-dark(drop-shadow(0 0 6px red), drop-shadow(0 0 4px pink))`, but it actually has to be `filter: drop-shadow(0 0 6px light-dark(red, pink))` if you can keep the same lengths, and otherwise you need to fall back to `@media (prefers-color-scheme: dark|light)`.

Using `light-dark` with non-colors still work if using e.g. lightningcss due to [transpilation](https://lightningcss.dev/transpilation.html#light-dark()-color-function), and thus the error might silently appear later. This can surface when changing chrome target to 123 since lightningcss stops transpiling light-dark due to native support.

[Lightningcss-demo](https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Atrue%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A7995392%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22.foo%20%7B%5Cn%20%20filter%3A%20light-dark(%5Cn%20%20%20drop-shadow(0%200%206px%20red)%2C%5Cn%20%20%20drop-shadow(0%200%202px%20green))%3B%5Cn%5Cn%7D%5Cn%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D)

### What solution would you like to see?

Warning/error about using light-dark where the arguments are not css-colors or css-variables.

**Comment by henrikhorluck** [comment]
There is some discussions about extending `light-dark` to work for non-colors or using `if` as an alternative: https://github.com/w3c/csswg-drafts/issues/10577, https://github.com/w3c/csswg-drafts/issues/12513.

The error/warning should perhaps recommend using 
```css
if(
  media(prefers-color-scheme: dark): drop-shadow(0 0 4px pink);
  else: drop-shadow(0 0 6px red);
);
 ``` 
instead? Though browser support for that is sub-optimal as of yet https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Values/if

**Comment by romainmenke** [comment]
Hi @henrikhorluck,

We give feedback for issues of this kind with the [`declaration-property-value-no-unknown`](https://stylelint.io/user-guide/rules/declaration-property-value-no-unknown) rule.

It seems that the syntax for `filter` specifically is very permissive.
This is currently defined as `none | <filter-value-list> | <-ms-filter-function-list>` where `-ms-filter-function-list` allows: `<ident-token> | <function-token> <any-value>? )`.

This implies that any keyword or any function is considered valid.

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCEAEwA6A7DGKAlgDYAuMATvBiUQOYAWZAtFAIaUDWAFGhABoMAI04BKANx4CxclRpQIMAM4tcEVjAAeRZWR6S8AXzwgBIWTABy7ALZxE2uwAcSMJGGXKz4CLmL0CCDazhCUZBiwKOwAruRY0hiUcSo0OPgEGNggsGAknOxkRH4szpQQzlRkAJ4sAG7sJDEwahAsMbhc6gDuuNk0ZMkwiSa4RlK4PpD+DABiYbaFQXrVbnS4ZB5+AUi2AFbe5rDO3ojpBNkra0Qb-Vkg+RR62QKJlzXXGyzTASx67LgOJQoHdso8VGRssZTEYgA)

-----

Compare this with an updated syntax:

```js
export default {
  rules: {
    "declaration-property-value-no-unknown": true
  },
  languageOptions: {
    syntax: {
      properties: {
        "filter": "none | <filter-value-list>"
      }
    }
  }
};
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCEAEwA6A7DGKAlgDYAuMATvBiUQOYAWZAtFAIaUDWAFGhABoMAI04BKANx4CxclRpQIMAM4tcEVjAAeRZWR6S8AXzwgBIWTABy7ALZxE2uwAcSMJGGXKz4CLmL0CCDazhCUZBiwKOwAruRY0hiUcSo0OPgEGNggsGAknOxkRH4szpQQzlRkAJ4sAG7sJDEwahAsMbhc6gDuuNk0ZMkwiUYCifm49DHs9DAA8s5FfsppiQTK1bhk7FqrGZkYZRVVRKkJ+wdZFqQUlP1X6rgwGAA+GAA8srf1jc0sdHoAHzZNaZEwXcEEcFGKS4HyQfwMABiYVshSCemqbjoWw8fgCSFsACtvOZYM5vIh0gRspjsUQtvdsvkKHpsmMMrSavStiwEQEWHp2LgOJQoEyQCyVGQQbhwSAjEA)

I think I am going to go with the type that is currently set in the specification: `none | <filter-value-list>` instead of what is currently set in `css-tree`: `none | <filter-value-list> | <-ms-filter-function-list>`.

**Comment by romainmenke** [comment]
@henrikhorluck,

The syntax in `@csstools/css-syntax-patches-for-csstree` has been updated and such values in the `filter` property will now error.

Updating your dependencies should pull in this version.

**Comment by henrikhorluck** [comment]
Thanks! I submitted a PR to the upstream csstree as well but great to hear it has been fixed in the patches already
